cdc_replication_test.py

```diff
@@ -0,0 +1,203 @@
+from cassandra.query import SimpleStatement
+from cassandra.cluster import Cluster
+
+from sdcm import cluster
+from sdcm.tester import ClusterTester
+
+def print_file_to_stdout(path: str) -> None:
+    with open(path, "r") as f:
+       shutil.copyfileobj(f, sys.stdout)
+
+# Write a CQL select result to a file.
+def write_cql_result(res, path: str):
+    with open(path, 'w') as f:
+        for r in res:
+            f.write(str(r) + '\n')
+        f.flush()
+        os.fsync(f.fileno())
+
+SCYLLA_MIGRATE_URL = "https://kbr-scylla.s3-eu-west-1.amazonaws.com/scylla-migrate"
+REPLICATOR_URL = "https://kbr-scylla.s3-eu-west-1.amazonaws.com/scylla-cdc-replicator-0.0.1-SNAPSHOT-jar-with-dependencies.jar"
+
+class CDCReplicationTest(ClusterTester):
+    KS_NAME = 'ks1'
+    TABLE_NAME = 'table1'
+
+    def collect_data_for_analysis(self,
+            migrate_log_path: str, replicator_log_path: str,
+            master_node: cluster.BaseNode, replica_node: cluster.BaseNode) -> None:
+        self.log.info('scylla-migrate log:')
+        print_file_to_stdout(migrate_log_path)
+        self.log.info('Replicator log:')
+        print_file_to_stdout(replicator_log_path)
+
+        with self.cql_connection_patient(node=master_node) as sess:
+            self.log.info('Fetching master table...')
+            res = sess.execute(SimpleStatement(f'select * from {self.KS_NAME}.{self.TABLE_NAME}',
+                consistency_level=ConsistencyLevel.QUORUM, fetch_size=1000))
+            write_cql_result(res, os.path.join(self.logdir, 'master-table'))
+
+        with self.cql_connection_patient(node=replica_node) as sess:
+            self.log.info('Fetching replica table...')
+            res = sess.execute(SimpleStatement(f'select * from {self.KS_NAME}.{self.TABLE_NAME}',
+                consistency_level=ConsistencyLevel.QUORUM, fetch_size=1000))
+            write_cql_result(res, os.path.join(self.logdir, 'replica-table'))
+
+    def test_replication(self) -> None:
+        self.log.info('Waiting for the latest CDC generation to start...')
+        # 2 * ring_delay (ring_delay = 30s) + leeway
+        time.sleep(70)
+
+        self.log.info('Starting stressor.')
+        stress_thread = self.run_stress_cassandra_thread(stress_cmd=self.params.get('stress_cmd'))
+
+        self.log.info('Let stressor run for a while...')
+        # Wait for C-S to create keyspaces/tables/UTs
+        time.sleep(20)
+
+        self.log.info('Fetching schema definitions from master cluster.')
+        master_node = self.db_cluster.nodes[0]
+        master_cluster = Cluster([master_node.external_address])
+        master_cluster.control_connection_timeout = 20
+        # need to connect so the driver can fetch schema metadata,
+        # even though we don't need the session itself:
+        with master_cluster.connect() as _:
+            master_cluster.refresh_schema_metadata()
+            ks = master_cluster.metadata.keyspaces[self.KS_NAME]
```

Comment:
I think better use standard sct methods for connection, this allow fix any connection issue without patching the test.
These will make sense, when cassandra driver will be updated and will not allow to use list of nodes in constructor for Cluster class.
for example next could be use:
```
        with self.cql_connection_patient_exclusive(node=master_node) as scylla_session:
            scylla_session.cluster.refresh_schema_metadata()
            ks = scylla_session.cluster.metadata.keyspaces[self.KS_NAME]
```

Reply:
Thanks, I was wondering how to get the `Cluster` object.